### PR TITLE
[docs] support YT videos in Video component

### PR DESCRIPTION
**Pull request by Simek**
# Why

Currently, we use the `Video` component which is based on the `react-player` for displaying the static `mp4`, when YouTube videos are embedded using `object` tag.

# How

`react-player` supports playing YouTube videos out-of-the-box, but we were using the trimmed down version just for local files, so to unify how we embedded videos I just had to correct import and adjust the empty value according to the types.

With the YT support, I have replaced all `object` and `embedded` tags on the pages with our Video component.

Additionally, on one of the page I have added `ImageSpotlight` for the images just inserted inline.

# Test Plan

The changes have been tested by running docs website locally.

## Preview - local video

Screenshot 2022-09-21 131119

## Preview - YT video

Screenshot 2022-09-21 131040

# Checklist

- [x] Documentation is up to date to reflect these changes (eg: https://docs.expo.dev and README.md).
- [x] Conforms with the [Documentation Writing Style Guide](https://github.com/expo/expo/blob/main/guides/Expo%20Documentation%20Writing%20Style%20Guide.md)
- [ ] This diff will work correctly for `expo prebuild` & EAS Build (eg: updated a module plugin).

### CODE REVIEW

**Comment by amandeepmittal** [comment]
@Simek, is it okay for videos to not show controls like play/pause button or play any audio if there is audio to a video such as on https://docs.expo.dev/development/introduction/?

Also, the video plays automatically when I go to that page. Is that desirable behavior for the YT videos?

**Comment by Simek** [comment]
So far, for the local files, we were additionally hiding all controls before first interaction, which probably is not ideal for the YT, since then there is no control over volume or seek. To improve this, I have altered the logic, so the full controls are always visible for YT videos, nice catch! 👍 

And answering the second part, this is how we play videos in general, they start playing muted when user scroll the video container into the viewport. Don't have a strong opinion on this pattern, but I feel like it's nice to have an unified behavior. What are your thoughts?

File: docs/components/plugins/Video.tsx [hunk 1]
```diff
@@ -46,7 +46,7 @@ type VideoProps = React.PropsWithChildren<{
 
 const Video = ({ controls, spaceAfter, url, file, loop = true }: VideoProps) => {
   const [hover, setHover] = useState(false);
-  const [forceShowControls, setForceShowControls] = useState(false);
+  const [forceShowControls, setForceShowControls] = useState(url?.includes('youtube.com'));
```
Comment (fobos531) [inline docs/components/plugins/Video.tsx:49]: Will this logic handle youtu.be links? I'm sure the `react-player` will internally load the video just fine, but I'm wondering if the controls will still be shown if we're relying on the explicit "youtube.com" string from the `url` prop for that 🤔

**Review by amandeepmittal — APPROVED** [review]
Thanks for including the player controls for YT videos 🙏

File: docs/components/plugins/Video.tsx (hunk 1 above)
Comment (Simek) [reply docs/components/plugins/Video.tsx:49]: We are currently not using the shortened YT links (and an URL need to be provided manually by editor), but probably we can add this additional domain to the check.

@fobos531 Are you able to add code suggestion in this PR, if not, would you like to make a follow up PR adding that check? 🙂

Comment (fobos531) [inline docs/components/plugins/Video.tsx:49]:
```suggestion
  const ytDomains = ['youtube.com', 'youtu.be'];
  const [forceShowControls, setForceShowControls] = useState(ytDomains.some((domain) => url?.includes(domain)));
```

**Review by fobos531 — CHANGES_REQUESTED** [review]

**Pull request merged**